components/raftstore/src/coprocessor/mod.rs

```diff
@@ -75,7 +75,7 @@ pub trait AdminObserver: Coprocessor {
     fn pre_apply_admin(&self, _: &mut ObserverContext<'_>, _: &AdminRequest) {}
 
     /// Hook to call after applying admin request.
-    fn post_apply_admin(&self, _: &mut ObserverContext<'_>, _: &mut AdminResponse) {}
+    fn post_apply_admin(&self, _: &mut ObserverContext<'_>, _: &AdminResponse) {}
```

Comment:
It's on purpose to allow modifying the response.

Reply:
There is no observer implement this method, and I think it is risky to take `&mut` since there is no mechanism to prevent two `Coprocessor` mistakenly modify the same field.

Reply:
> prevent two Coprocessor mistakenly modify the same field.

Two coprocessors should be allow to do so. It's coprocessor and callers of registration's responsibility to maintain correct ordering. For other methods, a bool is returned to allow skipping following all coprocessors.

Reply:
Yes, but I think it is unnecessary to use `&mut` here since there is no such use case.